controllers/metal3.io/hostfirmwaresettings_controller.go

```diff
@@ -120,6 +120,9 @@ func (r *HostFirmwareSettingsReconciler) Reconcile(ctx context.Context, req ctrl
 	bmh := &metal3v1alpha1.BareMetalHost{}
 	if err = r.Get(context.TODO(), req.NamespacedName, bmh); err != nil {
 		reqLogger.Info("could not get baremetalhost, not running reconciler")
+		if k8serrors.IsNotFound(err) {
+			return ctrl.Result{}, nil
```

Comment:
should not return error not found instead of nil ?

Reply:
Not completely sure, but we don't set any error for the BMH controller either if it is missing: https://github.com/metal3-io/baremetal-operator/blob/main/controllers/metal3.io/baremetalhost_controller.go#L127

Reply:
/lgtm

Reply:
returning an error would cause an immediate requeue, so this is correct.